src/main/java/fi/otavanopisto/kuntaapi/server/integrations/vcard/VCardTranslator.java

```diff
@@ -0,0 +1,271 @@
+import ezvcard.property.StructuredName;
+import ezvcard.property.Telephone;
+import ezvcard.property.TextProperty;
+import fi.metatavu.kuntaapi.server.rest.model.Address;
+import fi.metatavu.kuntaapi.server.rest.model.Contact;
+import fi.metatavu.kuntaapi.server.rest.model.ContactPhone;
+import fi.metatavu.kuntaapi.server.rest.model.ContactStatus;
+import fi.metatavu.kuntaapi.server.rest.model.LocalizedValue;
+import fi.otavanopisto.kuntaapi.server.id.ContactId;
+import fi.otavanopisto.kuntaapi.server.integrations.vcard.mecm.Status;
+
+@ApplicationScoped
+@SuppressWarnings ("squid:S3306")
+public class VCardTranslator {
+  
+  private static final String MECM_ADDITIONAL_STATUS_JSON = "mecm-status-json";
+  
+  @Inject
+  private Logger logger;
+  
+  public Contact translateVCard(ContactId kuntaApiContactId, VCard vCard) {
+    Contact result = new Contact();
+    
+    result.setAdditionalInformations(getAll(vCard.getNotes()));
+    result.setAddresses(translateAddresses(vCard.getAddresses()));
+    result.setDisplayName(getString(vCard.getFormattedName()));
+    result.setEmails(getAll(vCard.getEmails()));
+    result.setFirstName(getFirstName(vCard.getStructuredName()));
+    result.setId(kuntaApiContactId.getId());
+    result.setLastName(getLastName(vCard.getStructuredName()));
+    result.setOrganization(getOrganization(vCard.getOrganization()));
+    result.setOrganizationUnits(getOrganizationUnits(vCard.getOrganizations()));
+    result.setPhones(translatePhones(vCard.getTelephoneNumbers()));
+    result.setStatuses(translateStatuses(vCard));
+    result.setTitle(getFirst(vCard.getTitles()));
+    
+    return result;
+  }
+  
+  private List<ContactStatus> translateStatuses(VCard vCard) {
+    // Statuses are currenctly supported only in MECM flavoured vCards
+    
+    List<RawProperty> properties = vCard.getExtendedProperties(MECM_ADDITIONAL_STATUS_JSON);
+    if (properties != null) {
+      List<ContactStatus> translateMecmStatuses = new ArrayList<>(properties.size());
+      for (RawProperty property : properties) {
+        translateMecmStatuses.add(translateMecmStatus(property.getValue()));
+      }
+      
+      return translateMecmStatuses;
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private ContactStatus translateMecmStatus(String value) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.registerModule(new JavaTimeModule());
+    
+    try {
+      Status mecmStatus = objectMapper.readValue(value, Status.class);
+      
+      ContactStatus contactStatus = new ContactStatus();
+      contactStatus.setEnd(mecmStatus.getArrival());
+      contactStatus.setStart(mecmStatus.getDeparture());
+      contactStatus.setText(mecmStatus.getReason());
+      
+      return contactStatus;
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Failed to parse MECM statuses", e);
+    }
+    
+    return null;
+  }
+
+  private List<ContactPhone> translatePhones(List<Telephone> vCardNumbers) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translatePhones" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)